### PR TITLE
Editor as SPA

### LINKED ISSUE #787: Editor as SPA
Editor will be SPA without server-side:
* remove server side
  - [x] redirect all requests to api/authoring
  - [ ] remove web-push logic
  - [x] implement editor-configuration logic
  - [ ] remove service worker logic
  - [ ] implement version endpoint in secure-gateway
* implement openid connect
  - [x] implement in editor
  - [x] implement in secure-gateway
* implement simple authentication protocol (digest-like)
  - [ ] change app creator in authoring to sync apps to minio
  - [ ] add endpoint to secure-gateway to create apps

### PR DESCRIPTION
**Pull request by AleF83**
Closes #787

### CODE REVIEW

File: services/publishing/Tweek.Publishing.Service/Sync/Converters/ExternalAppsConverter.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+namespace Tweek.Publishing.Service.Sync.Converters
+{
+    public class ExternalAppsConverter : IConverter
+    {
+        private static readonly Regex externalAppRegex = new Regex(Patterns.ExternalApp, RegexOptions.Compiled);
+        private readonly IObjectStorage _client;
+
+        public ExternalAppsConverter(IObjectStorage storageClient)
+        {
+            _client = storageClient;
```
Comment (Yshayy) [inline services/publishing/Tweek.Publishing.Service/Sync/Converters/ExternalAppsConverter.cs]: _client is not used

File: services/publishing/Tweek.Publishing.Service/Sync/Converters/ExternalAppsConverter.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    }
+                });
+            result.Add(appId, app);
+
+            return (@"external_apps.json", JsonConvert.SerializeObject(result), @"application/json");
+        }
+
+        private static string CreateSecretKey()
+        {
+            var key = Environment.GetEnvironmentVariable("GIT_SERVER_PRIVATE_KEY_PATH");
```
Comment (Yshayy) [inline services/publishing/Tweek.Publishing.Service/Sync/Converters/ExternalAppsConverter.cs]: It's probably better to inject the key itself in the ctor.

File: services/publishing/Tweek.Publishing.Service/Model/ExternalApps/ExternalApp.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Tweek.Publishing.Service.Model.ExternalApps
+{
+    public class ExternalApp
+    {
+        [JsonProperty("name")]
+        public string Name;
+
+        [JsonProperty("version")]
+        public string Version;
```
[the remark this answers is not in the document]
Comment (AleF83) [reply services/publishing/Tweek.Publishing.Service/Model/ExternalApps/ExternalApp.cs]: This is field in AuthoringApi https://github.com/Soluto/tweek/blob/d361f14458b88bda81fa07c978801bdac2607442/services/authoring/src/routes/apps.ts#L33

**Comment by michaelkruglos** [comment]
We need to do something REACT_APP_GATEWAY_URL build arg for editor. It should be solved somehow else.

**Pull request merged**